src/Common/Commands.Common/CustomAttributes/BreakingChangeAttributeHelper.cs

```diff
@@ -0,0 +1,151 @@
+    ## Current Breaking Changes
+
+    The following cmdlets were affected this release:
+
+    **Cmdlet 1**
+    - Description of what has changed
+
+    ```powershell
+    # Old
+    # Sample of how the cmdlet was previously called
+
+    # New
+    # Sample of how the cmdlet should now be called
+    ```
+
+    ## Release X.0.0
+
+    The following cmdlets were affected this release:
+
+    **Cmdlet 1**
+    - Description of what has changed
+
+    ```powershell
+
+    # Old\
+    # Sample of how the cmdlet was previously called
+
+    # New
+    # Sample of how the cmdlet should now be called
+    ```
+
+    Note: the above sections follow the template found in the link below:
+
+    https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes/breaking-change-template.md
+    -->";
+
+        public static void processCustomAttributesAtRuntime(Type type, List<String> boundParameterNames)
+        {
+            bool emitWarningOrError = true;
+
+            try
+            {
+                emitWarningOrError = Boolean.Parse(System.Environment.GetEnvironmentVariable(SUPPRESS_ERROR_OR_WARNING_MESSAGE_ENV_VARIABLE_NAME));
+            }
+            catch (Exception)
+            {
+                //no action
+            }
+
+            if (!emitWarningOrError)
+            {
+                //Do not process the attributes at runtime... The env variable to override the warning messages is set
+                return;
+            }
+
+            List<GenericBreakingChangeAttribute> attributes = getAllBreakingChangeAttributesInType(type, boundParameterNames);
+
+            if (attributes != null && attributes.Count > 0)
+            {
+                Console.WriteLine("Breaking changes in the cmdlet : " + Utilities.getNameFromCmdletType(type) + "\n");
+            }
+
+            foreach (GenericBreakingChangeAttribute attribute in attributes)
+            {
+                attribute.printCustomAttributeInfo(type, false);
+            }
+        }
+
+        public static List<String> getBreakingChangeMessagesForType(Type type)
+        {
+            List<String> messages = new List<string>();
+
+            //This is used as a migration guide, we need to process all properties/fields, moreover at this point of time we do not have a list of all the
+            //bound params anyways
+            foreach (GenericBreakingChangeAttribute attribute in getAllBreakingChangeAttributesInType(type, null))
+            {
+                messages.Add(attribute.getBreakingChangeTextFromAttribute(type, true));
+            }
+
+            return messages;
+        }
+
+        public static List<GenericBreakingChangeAttribute> getAllBreakingChangeAttributesInType(Type type, List<String> boundParameterNames)
+        {
+            List<GenericBreakingChangeAttribute> attributeList = new List<GenericBreakingChangeAttribute>();
+
```

Comment:
In general, the .Net rules for vertical space are that there should be vertical space after a closing bracket ```}```, but generally not in most other cases - tighten up this vertical spacing and be somewhat cognizant of line breaks to improve readability.